### PR TITLE
Project

### PR DESCRIPTION
**Pull request by Virtruvian1**

### CODE REVIEW

**Comment by Virtruvian1** [comment]
Hey guys, I'm sorry the CSV Parser was put on the backend this week, I was really looking into alot of different jobs in the Knoxville, TN area. I did read into it and got a solid plan laid out for it which I believe won't take that long to code through.

**Comment by Virtruvian1** [comment]
There's a CSVParserPlan.txt file in the main repository that shows the details to the plan that I laid out, and if you swap into the Program.CS you can see the code I started with. I can share a Visual Studio link and we can work on it on one file if you guys want.

**Comment by taylor-d-jackson** [comment]
No problem Zach.

File: CSV Parse/Program.cs [hunk 1]
```diff
@@ -1,18 +1,73 @@
...
 
 namespace Program
 {
+    public class ParserEngine
+    {
+        private string _filePath;
+        private string _currentColumn;
+        private string _currentTable;
+
+        public ParserEngine() // Default Constructor
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:17]: Default Constructor set to initialize the default values

**Comment by taylor-d-jackson** [comment]
Sounds good.

Respectfully,

Taylor D. Jackson
Software Developer
(949) 302-4505
taylor.daniel.jackson@outlook.com
LinkedIn Profile<https://www.linkedin.com/in/taylordjackson/>
________________________________

File: CSV Parse/Program.cs [hunk 2]
```diff
@@ -1,18 +1,73 @@
...
+        {
+            _filePath = "Empty Path";
+            _currentColumn = "No Data";
+            _currentTable = "No Data";
+        }
+
+        public string FilePath // Checks FilePath
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:24]: I'm using 'get-set' functions for organization to hide the Exceptions and the test the data before we set it to private variables

File: CSV Parse/Program.cs [hunk 3]
```diff
@@ -1,18 +1,73 @@
...
+            get { return _filePath;  } 
+            set 
+            {
+                bool filePathExist = false; 
+                bool filePathCorrect = false;
+                int processCounter = 1;
+
+                switch (processCounter)
+                {
+                    case 1:
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:35]: Test 1 to ensure the file exist

File: CSV Parse/Program.cs [hunk 4]
```diff
@@ -1,18 +1,73 @@
...
+                        // Test 1
+                        filePathExist = File.Exists(FilePath);
+                        processCounter++;
+                        break;
+                    case 2:
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:40]: Test 2 to ensure the file  path leads to a .CSV file

File: CSV Parse/Program.cs [hunk 5]
```diff
@@ -1,18 +1,73 @@
...
+                        // Test 2
+                        processCounter++;
+                        break;
+                    case 3:
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:44]: Test 3 to check if Test 1 & 2 are true before setting the _filePath variable

File: CSV Parse/Program.cs [hunk 6]
```diff
@@ -1,18 +1,73 @@
...
+                        }
+                        else { Console.WriteLine("ERROR! The file path provided does not exist or is not a .CSV File, please try again"); }
+                        break;
+                    default:
+                        break;
+                }
+            } 
+        } 
+
+        public void TakeFilePath (string tempFilePath)
```
Comment (Virtruvian1) [inline CSV Parse/Program.cs:58]: Takes user input to get the file path

**Pull request merged**